aten/src/ATen/core/type.cpp

```diff
@@ -1086,6 +1087,28 @@ void ClassType::addMethod(torch::jit::Function* method) {
   methods_.push_back(method);
 }
 
+void ClassType::addOverloadedMethod(torch::jit::Function* method) {
+  // we can't use old findMethod because it searches based on the string name of
+  // function
+  for (auto added_method : methods_) {
+    if (method == added_method) {
+      return;
+    }
+  }
+
+  auto it = overloaded_methods_.insert(
+      std::pair<std::string, std::vector<std::string>>(
+          method->name(), std::vector<std::string>()));
+  // create a mangled name for this function and bookkeep
+  // the mangled name and its corresponding function.
+  const std::string& mangled_name =
+      method->name() + "__" + c10::guts::to_string(it.first->second.size());
+  it.first->second.push_back(mangled_name);
+  // registers where this overloaded function is stored in the methods map.
+  mangled_to_function_[mangled_name] = methods_.size();
```

Comment:
sure

Reply:
We don't need this for addMethod because it internally calls findMethod which makes sure if there is any overloaded/mangled/normal method with the given name.